pkg/packageinstall/packageinstall_version_handler.go

```diff
@@ -105,6 +105,7 @@ func (ipvh *PackageInstallVersionHandler) isEligibleForVersionUpgrade(version st
 
 	semverConfig := installedPkg.Spec.PackageRef.VersionSelection
 
+	// TODO: we should  make this apply the same kc and k8s constraints as the other one to avoid unnecessary reconciles
```

Comment:
So this watcher doesn't get the PKGI resource,  only the PKG. This means:

- We can't check for the absense / presense of the ignore annotations
- We also have to duplicate / make public the constraint checking functions
- It introduces more complexity into this version handler and another API call / cross-reference check

Will probably want to chat this one through as a group